File: services/client/src/protocol/packet/bet_packet.go
package packet

import (
	"github.com/7574-sistemas-distribuidos/tp-nivelador/src/lottery"
)

type BetPacket struct {
	Bets []lottery.Bet
}

func NewBetPacket(bets []lottery.Bet) Packet {
	return &BetPacket{Bets: bets}
}

func betFromBytes(payload []byte) (Packet, error) {
	offset := 0
	var bets []lottery.Bet
	for offset < len(payload) {
		bet, consumed, err := deserializeBet(payload[offset:])
		if err != nil {
			return nil, err
		}
		offset += consumed
		bets = append(bets, bet)
	}

	return NewBetPacket(bets), nil
}

func (bPkt *BetPacket) ToBytes() ([]byte, error) {
	var buff []byte

	for _, bet := range bPkt.Bets {
		payload, err := serializeBet(bet)
		if err != nil {
			return nil, err
		}
		buff = append(buff, payload...)
	}

	return buildFrame(BET_OPCODE, buff), nil
}

File: services/server/src/protocol/packet/bet_packet.py
from .packet import Packet, build_frame
from lottery import Bet
from .bet_codec import encode_bet, decode_bet
from .opcode import Opcode


class BetPacket(Packet):
    def __init__(self, bets: list[Bet]):
        self.bets = bets

    def to_bytes(self) -> bytes:
        payload = b"".join(encode_bet(bet) for bet in self.bets)
        return build_frame(Opcode.BET, payload)

def bet_pkt_from_bytes(data: bytes, agency_id: int) -> Packet:
    offset = 0
    bets = []

    while offset < len(data):
        bet, consumed = decode_bet(data[offset:], agency_id)
        bets.append(bet)
        offset += consumed

    return BetPacket(bets)
